R/bootstrap.nlsfit.R

```diff
@@ -1,100 +1,47 @@
-    ipx <- length(par)-seq(nx-1,0)
-    -as.vector(t(crossprod(dfitfun(par=par[-ipx], x=x[ipx], ...), dy)))
-    ## pieces missing
-  }
-
-  ## the corresponding chisqr functions
-  fitchisqr <- function(par, x, y, dy, fitfun, dfitfun, ...) {
-    z <- fitchi(par=par, x=x, y=y, dy=dy, fitfun=fitfun, dfitfun=NULL, ...)
-    return (sum(z * z))
-  }
-  dfitchisqr <- function(par, x, y, dy, dfitfun, fitfun, ...) {
-    z <- fitchi(par=par, x=x, y=y, dy=dy, fitfun=fitfun, dfitfun=dfitfun, ...)
-    dz <- dfitchi(par=par, x=x, y=y, dy=dy, dfitfun=dfitfun, fitfun=fitfun, ...)
-    return(2*(z %*% array(dz, dim=c(length(x), length(par)))))
-  }
-  fitchisqr.xy <- function(par, y, dy, fitfun, nx, ...) {
-    z <- fitchi.xy(par=par, y=y, dy=dy, fitfun=fitfun, nx=nx, ...)
-    return (sum(z * z))
-  }
-
-  ## wrapper functions for apply
-  wrapper.lm <- function(y, par, fitfun, dfitfun, dy, x, ...) {
-    res <- list()
-    if(is.null(dfitfun))  res <- minpack.lm::nls.lm(par=par, fn=fitchi, y=y, fitfun=fitfun, dy=dy, x=x,
-                                                    control = nls.lm.control(ftol=1.e-8, ptol=1.e-8, maxiter=500), ...)
-    else res <- minpack.lm::nls.lm(par=par, fn=fitchi, y=y, fitfun=fitfun, jac=dfitchi, dy=dy, x=x, dfitfun=dfitfun,
-                                   control = nls.lm.control(ftol=1.e-8, ptol=1.e-8, maxiter=500), ...)
-    if( !(res$info %in% c(1,2,3) ) ){
-      cat(sprintf("Termination wrapper.lm reason of nls.lm res$info: %d\n", res$info))
+  if(missing(dy)){
+    if(!useCov && sim == "parametric"){
+      stop("dy has to be provided!")
     }
-    return (c(res$par, res$rsstrace[length(res$rsstrace)]))
-  }
-  wrapper.lm.xy <- function(y, par, fitfun, dy, nx, ...) {
-    res <- list()
-    res <- minpack.lm::nls.lm(par=par, fn=fitchi.xy, y=y, fitfun=fitfun, dy=dy, nx=nx,
-                              control = nls.lm.control(ftol=1.e-8, ptol=1.e-8, maxiter=500),
-                              ...)
-    if( !(res$info %in% c(1,2,3) ) ){
-      cat(sprintf("Termination wrapper.lm.xy reason of nls.lm res$info: %d\n", res$info))
+    if(missing(bsamples) && missing(CovMatrix)){
+      stop("dy, bsamples or CovMatrix has to be provided!")
     }
-    return (c(res$par, res$rsstrace[length(res$rsstrace)]))
   }
 
-  wrapper.optim <- function(y, par, fitfun, dy, x, dfitfun, ...) {
-    res <- NA
-    if(is.null(dfitfun)) res <- optim(par=par, fn=fitchisqr, y=y, method=c("BFGS"), fitfun=fitfun, dy=dy, x=x, ...)
-    else res <- optim(par=par, fn=fitchisqr, gr=dfitchisqr, y=y, method=c("BFGS"), fitfun=fitfun, dfitfun=dfitfun, dy=dy, x=x, ...)
-
-    return (c(res$par, res$value))
-  }
-  wrapper.optim.xy <- function(y, par, fitfun, dy, nx, ...) {
-    res <- optim(par=par, fn=fitchisqr.xy, y=y, method=c("BFGS"), fitfun=fitfun, dy=dy, nx=nx, ...)
-    return (c(res$par, res$value))
+  lm.avail <- require(minpack.lm)
+  if(parallel){
+    parallel <- require(parallel)
```

Comment:
this package needs to be added to the hadron package description.

Reply:
@j-ostmeyer also this should be at least commented on

Reply:
ping

Reply:
This is resolved, in the current state of the branch there is `parallel` in the `Suggests` of the `DESCRIPTION`.